docs/learn-phragmen.md

```diff
@@ -526,18 +526,68 @@ their candidates selected will also not have any stake in any of the elected can
+
+To minimize block computation time, the staking process is run as an
+[off-chain worker](https://substrate.dev/docs/en/conceptual/core/off-chain-workers). In order to
+give time for this off-chain worker to run, staking commands (bond, nominate, etc.) are not allowed
+in the last quarter of each era.
+
+These optimizations will not be covered in-depth on this page. For more details, you can view the
+[Rust implementation of elections in Substrate](https://github.com/paritytech/substrate/blob/master/frame/elections-phragmen/src/lib.rs),
+the
+[Rust implementation of staking in Substrate](https://github.com/paritytech/substrate/blob/master/frame/staking/src/lib.rs),
+or the `seqPhragménwithpostprocessing` method in the
+[Python reference implementation](https://github.com/w3f/consensus/tree/master/NPoS). If you would
+like to dive even more deeply, you can review the
+[W3F Research Page on Sequential Phragmén Method](https://research.web3.foundation/en/latest/polkadot/NPoS/4.%20Sequential%20Phragm%C3%A9n%E2%80%99s%20method.html).
+
+### Rationale for Minimizing the Number of Validators Per Nominator
+
+Paying out rewards for staking from every validator to all of their nominators can cost a
+non-trivial amount of chain resources (in terms of space on chain and resources to compute). Assume
+a system with 200 validators and 1000 nominators, where each of the nominators has nominated 10
+different validators. Payout would thus require `1000 * 10`, or 10,000 transactions. In an ideal
+scenario, if every nominator selects a single validator, only 1,000 transactions would need to take
+place - an order of magnitude fewer. Empirically, network slowdown at the beginning of an era has
+been seen due to the large number of individual payouts by validators to nominators. In extreme
+cases, this could be an attack vector on the system, where nominators nominate many different
+validators with small amounts of stake in order to slow the system at the next era change.
+
+While this would reduce network and on-chain load, being able to select only a single validator
+incurs some diversification costs. If the single validator that a nominator has nominated goes
+offline or acts maliciously, then the nominator incurs a risk of a significant amount of slashing.
+Nominators are thus allowed to nominate up to 16 different validators. However, after the weighted
+edge-reducing algorithm is run, the number of validators per nominator is minimized. Nominators are
+likely to see themselves nominating a single active validator for an era.
+
+At each era change, as the algorithm runs again, nominators are likely to have a different validator
+than they had before (assuming a significant number of selected validators). Therefore, nominators
+can diversify against incompetent or corrupt validators causing slashing on their accounts, even if
+only nominate a single validator is nominated by them per era.
```

Comment:
```suggestion
only a single validator is nominated by them per era.
```

Reply:
Consider changing to active voice

Reply:
changed to ".. even if they only nominate a single validator per era."